lib/cudnn/multiheadattn.jl

```diff
@@ -40,8 +40,8 @@ Keyword arguments describing the attention operation when `d` is not given:
 
 Other keyword arguments:
 * `residuals = nothing`: optional tensor with the same size as queries that can be used to implement residual connections (see figure in cudnn docs). When residual connections are enabled, the vector length in `queries` should match the vector length in `out`, so that a vector addition is feasible. 
-* `currIdx::Integer = -1`: Time-step (0-based) in queries to process. When the currIdx argument is negative, all Q time-steps are processed. When currIdx is zero or positive, the forward response is computed for the selected time-step only. The latter input can be used in inference mode only, to process one time-step while updating the next attention window and Q, R, K, V inputs in-between calls.
-* `loWinIdx, hiWinIdx::Array{Cint}`: Two host integer arrays specifying the start and end (0-based) indices of the attention window for each Q time-step. The start index in K, V sets is inclusive, and the end index is exclusive. By default set at 0 and `kvMaxSeqLength` respectively.
+* `currIdx::Integer = -1`: Time-step (0-based) in queries to process. When the `currIdx` argument is negative, all ``Q`` time-steps are processed. When `currIdx` is zero or positive, the forward response is computed for the selected time-step only. The latter input can be used in inference mode only, to process one time-step while updating the next attention window and ``Q``, ``R``, ``K``, ``V`` inputs in-between calls.
```

Comment:
There is no previous mention of ``R``. I don't know this operation so I can't say whether ``R`` should just be removed here or documented elsewhere.

Reply:
Here is the original source: https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnMultiHeadAttnForward

After reading it again, I think you are correct: ``R`` is a typo and can be removed.

Reply:
I agree that ``R`` is definitely a typo but it seems plausible that it could be a typo for ``O``.

Reply:
I suspected the same thing. However in the more detailed description above (https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnMultiHeadAttnForward paragraph starting with "In the inference mode") it only mentions updating `Q,K,V` vectors. Also `O` is for output, i.e. the caller does not initialize it, only the return value is written over it.

Reply:
Ok, let's just remove it.